src/joinUrl.ts

```diff
@@ -0,0 +1,18 @@
+export const joinUrl = (...args: string[]): string => {
+	// Remove any leading or trailing slashes from all parts and then join cleanly on
+	// a single slash, preventing malformed urls
+	const trimmed = args
+		.map((part) => {
+			// Trim left
+			if (part.substr(0, 1) === '/') return part.slice(1);
+			return part;
+		})
+		.map((part) => {
+			// Trim right
+			if (part.substr(part.length - 1, 1) === '/')
+				return part.slice(0, -1);
+			return part;
+		});
```

Comment:
could you replace both of these with `.map(part => part.replace(/^\/|\/$/g, ''))`?

not sure what the perf implications are though, and only a style thing so not a blocker

Reply:
I could... but I fear that would leave the code harder to read - or at least to me it reads as a series of slashes and symbols.

I tend to reach for regexes when I _need_ them or if there's a strong performance gain from using them. Outside of this, I find javascript's own native methods a nicer set of tools to use.

That said, I'm not against them, maybe I just need to learn them better? And I'm not strongly opposed to changing this code here. It's also interesting that you raise it as a style consideration which means that not everyone has the same subjective ideas as me when it comes to how code should look (shocker!)

Reply:
yeah if you prefer it this way all good 👍

Reply:
> maybe I just need to learn them better?

things of beauty :)

Reply:
also regex way was 34% slower on jsbench.me so this has just been noise. apologies!